### PR TITLE
Backend JSON structure documentation

### PR DESCRIPTION
**Pull request by tadhg-ohiggins**
Need things be complex?
Apparently yes, for forms;
Let's try some structure.

Documentation for the backend JSON structure, and the JSON for Section 1 and Section 2b.
Addresses #240.

### CODE REVIEW

**Comment by tadhg-ohiggins** [comment]
It wasn't immediately evident to me how to get this link to see the documentation in a reasonable format, so: https://github.com/18F/cms-carts-seds/blob/backend-json-structure-docs/docs/backend-json-structure-documentation.rst

File: docs/backend-json-section-3.json [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            "type": "subsection",
+            "id": "2020-03-a",
+            "title": "Program Outreach",
+            "parts": [
+                {
+                    "id": "2020-03-a-01",
+                    "questions": [
+                        {
+                            "id": "2020-03-a-01-01",
+                            "text": "Have you changed your outreach methods in the last federal fiscal year?",
```
Comment (cannandev) [inline docs/backend-json-section-3.json]: Can we change this property from `text` to `label`?

File: docs/backend-json-section-3.json [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                            "type": "radio",
+                            "answer": {
+                                "options": {
+                                    "Yes": "yes",
+                                    "No": "no"
+                                },
+                                "entry": null
+                            },
```
Comment (cannandev) [inline docs/backend-json-section-3.json]: Can we add a flag if this question has previous year data? 
```
"entry": null,
"previous_year": true
```

File: docs/backend-json-section-3.json [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    "state": null,
+    "valid": null,
+    "ordinal": 3,
+    "id": "2020-03",
+    "type": "section",
+    "title": null,
+    "subsections": [
+        {
+            "type": "subsection",
+            "id": "2020-03-a",
```
Comment (cannandev) [inline docs/backend-json-section-3.json]: Add a the page url for this subsection? We would use this for routing in the main navigation.
```
"route": "section/3d",
```

**Review by cannandev — CHANGES_REQUESTED** [review]

File: docs/backend-json-section-2.json [hunk 4]
```diff
@@ -0,0 +1,200 @@
+{
+    "year": 2020,
+    "state": null,
+    "valid": null,
+    "ordinal": 2,
+    "type": "section",
+    "title": "Program Fees and Policy Changes",
```
[the remark this answers is not in the document]
Comment (tadhg-ohiggins) [reply docs/backend-json-section-2.json]: Fixed that, good catch.

File: docs/backend-json-section-3.json (hunk 1 above)
Comment (tadhg-ohiggins) [reply docs/backend-json-section-3.json]: Done.

File: docs/backend-json-section-3.json (hunk 2 above)
Comment (tadhg-ohiggins) [reply docs/backend-json-section-3.json]: I added `display_prior_year_data` as an allowed property in `context_data` in the schema, but I haven't added the actual indications about whether or not prior year's data should be displayed to the JSON data.

File: docs/backend-json-section-3.json (hunk 3 above)
Comment (tadhg-ohiggins) [reply docs/backend-json-section-3.json]: I'm unwilling to put routing in the data… if it ends up being really awkward not to, we can revisit this.

**Comment by tadhg-ohiggins** [comment]
@cannandev I addressed the comments, and also made lots of other updates. I think this is good to go, feel free to merge it.

**Review by jgillis1 — APPROVED** [review]

**Review by cannandev — APPROVED** [review]

**Pull request merged**